core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -3242,6 +3236,9 @@ protected function submit()
 
 			$this->invalidateCacheTags();
 		}
+
+		// Empty cached data
+		$this->arrCurrentRecordCache = array();
```

Comment:
Are we sure to delete **all** caches? And not just the one of the record we just submitted? Also, if we only clear one, we might want to do this at the top (before the `noReload` check)

Reply:
Fixed in 7dac6bfa7495dec3cc2e13399d47a70c21766044.
Why before? We need the cache for `getCurrentRecord()`.